Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/repo/ContactDiaryRepository.kt

```diff
@@ -0,0 +1,44 @@
+package de.rki.coronawarnapp.contactdiary.storage.repo
+
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocation
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocationVisit
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryPerson
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryPersonEncounter
+import kotlinx.coroutines.flow.Flow
+import org.joda.time.LocalDate
+
+@Suppress("TooManyFunctions")
+interface ContactDiaryRepository {
+
+    // Location
+    val locations: Flow<List<ContactDiaryLocation>>
+    suspend fun addLocation(contactDiaryLocation: ContactDiaryLocation)
+    suspend fun updateLocation(contactDiaryLocation: ContactDiaryLocation)
+    suspend fun deleteLocation(contactDiaryLocation: ContactDiaryLocation)
+    suspend fun deleteLocations(contactDiaryLocations: List<ContactDiaryLocation>)
+    suspend fun deleteAllLocations()
+
+    // Location visit
+    val locationVisits: Flow<List<ContactDiaryLocationVisit>>
+    fun locationVisitsForDate(date: LocalDate): Flow<List<ContactDiaryLocationVisit>>
+    suspend fun addLocationVisit(contactDiaryLocationVisit: ContactDiaryLocationVisit)
+    suspend fun deleteLocationVisit(contactDiaryLocationVisit: ContactDiaryLocationVisit)
```

Comment:
A single deletion method that accepts a list is sufficient.

Reply:
This method is being used and works fine. Why refactor?

Reply:
You both are right.
it is not really needed but its like a comfort feature that you not have to create a list for adding a simple single item.
Would let it as it is for now.